### ISSUE TITLE
Support ...$ in type arguments

### ISSUE THREAD

**Issue by xeno-by** [scalameta/scalameta#519]
It would be nice to be able to write `val q"$fn[...$targs](...$argss) = ...` to be able to use one quasiquote to match applications that contain or don't contain type arguments. If one writes,`..$targs`, then the match will crash on things like`foo(bar)`.

**Comment by blast-hardcheese** [comment]
Discussion in #602

**Comment by xeno-by** [comment]
It'd be great to use a similar approach to get rid of other `nel`s in quasiquote.md. See the commit linked above for more details.

**Comment by DavidDudson** [comment]
Linking this project https://github.com/alexknvl/newts/blob/master/src/main/scala/newts/opaque.scala

**Comment by mehmeteking** [comment]
I came across this 'issue' recently and the proposed solution sounds really nice.
Is there any progress in this area? If not, may I have a look at it?

**Comment by blast-hardcheese** [comment]
I wasn't able to go back and implement the suggestion. I've got more shims than I'd like around this, so I'd be super grateful if you took over!

**Comment by xeno-by** [comment]
@olafurpg and I had a discussion, and we decided to pronounce quasiquotes feature-complete.

There are still some open issues (#46, #223, #251, #280 and others), but they all have two things in common: 1) they require non-trivial design work to be fixed, 2) they aren't popular at all. Given our recent decision to start avoiding unwarranted breaking changes in 2.0, we'll consider quasiquotes good enough at this point and will close all outstanding feature requests.

We will still be fixing bugs (in fact, I've fixed about a dozen of such bugs for 2.0), but without strong evidence our design efforts will be directed elsewhere.
